domain/model/service/modelservice.go

```diff
@@ -94,5 +107,58 @@ func (s *ModelService) CreateModel(
 func (s *ModelService) DeleteModel(
 	ctx context.Context,
 ) error {
-	return s.st.Delete(ctx, s.modelID)
+	return s.modelSt.Delete(ctx, s.modelID)
+}
+
+// Status returns the status of the model.
+func (s *ModelService) Status(ctx context.Context) (model.StatusInfo, error) {
+	modelState, err := s.controllerSt.GetModelState(ctx, s.modelID)
+	if err != nil {
+		return model.StatusInfo{}, err
+	}
+	if modelState.Destroying {
+		return model.StatusInfo{
+			Status:  corestatus.Destroying,
+			Message: "the model is being destroyed",
```

Comment:
We don't seem to be setting the time field on any off these computed status values.

Reply:
No, because we don't know the time of the status change.
If you can figure out where we can find it, feel free to include this time in the result returned from the state.

Reply:
Are you sure that is what we want to do? Should we just not set `time.Now()` at least that has meaning over `time.Zero()`.

If we tell the client to start doing is == time.Zero it means we have changed the contract of the api and we should technically bump the apiserver facade version.


Keep it simple I say.